xarray/core/dataset.py

```diff
@@ -1111,7 +1144,9 @@ def _replace(
         """
         if inplace:
             if variables is not None:
-                self._variables = variables
+                self._manifest.variables = variables
+                # TODO if ds._variables properly pointed to ds._manifest.variables we wouldn't need this line
+                self._variables = self._manifest.variables
```

Comment:
What I wanted was for `self._variables` to always point to `self._manifest.variables`, such that updating the former by definition updates the latter. But I'm not really sure if that kind of pointer-like behaviour is possible.